src/common/acl_principal.c

```diff
@@ -298,7 +298,7 @@ daos_acl_principal_to_gid(const char *principal, gid_t *gid)
 	if (rc == -DER_NOMEM)
 		D_GOTO(out, rc);
 	if (rc != 0) {
-		D_ERROR("Error from getgrnam_r: %d\n", rc);
+		D_ERROR("Error from getgrnam_r: "DF_RC"\n", DP_RC(rc));
```

Comment:
It appears rc would be normal errno here (and other places in the file).  Any time we call d_errno2der, we would want to print the regular errno or do a print after the conversion but never both.

Reply:
good point, will double check the patch for errno2der and der2errno cases make sure the msgs are proper and clean.

Reply:
The issue here for such a big patch is that it's not obvious what type of rc is used by each function, seems like it's mixing up now, have to dig into each function hierarchical calls to figure that out if possible. Any pattern can be used?

Reply:
Not really.   The nice thing about DF_RC is that it will at least print out the numeric code as well as the string.  We may get some false strings in such cases but there probably aren't that many of them.   In this case it will be obvious when a bug is reported and can ultimately be changed then.